src/mon/OSDMonitor.cc

```diff
@@ -7699,7 +7699,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       pool = CrushWrapper::DEFAULT_CHOOSE_ARGS;
       positions = 1;
     }
-    newcrush.create_choose_args(pool, positions);
+    if (!newcrush.create_choose_args(pool, positions)) {
+      ss << "requested weight-set already created";
```

Comment:
i'd suggest reword the error message, like:
```c++
ss << "weight-set for pool '" << poolname << "' already created";
```

because we don't specify any certain weight-set here.